### DISCUSSION TITLE
Postgres connection won't connect

### DISCUSSION THREAD

**Discussion by binaryben** [supabase/supabase#13888] [Questions] [answered]
I've set up a project on the free tier. It seemed to be working well but has suddenly stopped database connections. Here is a sample from a psql connection that has worked fine previously.

```
psql: error: connection to server at "db.[id].supabase.co" ([ip]), port 5432 failed: Connection refused
	Is the server running on that host and accepting TCP/IP connections?
```

Sent a support ticket but understandably got an email saying free users should come here. Any ideas?

I'd set up two projects and got an email saying one was paused due to inactivity (no problems, as expected). No emails for this one though.

**Comment by binaryben** [comment, marked as answer]
I misconfigured a new service and used the host as the username. I suspect there must be a ban period for failed attempts to stop brute force attacks. I locked my IP out as it started working again after some time for existing services, and then failed with the same error when I tried to run the new service again.

EDIT: The team answered the email anyway and confirmed. In case anyone else finds this and needs to know, your IP will be blocked for 30 mins after two failed attempts to login to a database. You can unban your IP using the CLI (neat!)

**Reply by jjohnson5253** [reply]
How do you unban your IP? I tried using cli but got this error:
```Error: failed to retrieve network restrictions; received: {"message":"project <projectRef> not allowed to set up network restrictions; please contact Supabase support."}```

**Reply by binaryben** [reply]
@jjohnson5253 Not sure how to unblock via the CLI as I haven't had to since I originally posted. But I think the ban only lasts 30 mins from memory, so hopefully you have access again

**Reply by rimzudev** [reply]
Thank you very much. That helped me a lot!

To get the unbanned IPs:
```supabase network-bans get --experimental```
To unban an IP:
```supabase network-bans remove --db-unban-ip $PASTE_IP_HERE --experimental```

_You may have to add "npx" at the front if you are using the CLI with npm. The experimental flag may be removed in the future._

**Reply by bajahranks** [reply]
You can now unban via the supabase UI. Go to your project settings and then go to database. Scroll all the way to the bottom.

**Reply by krugerm** [reply]
unban IP only supports IPv4, so if you're using IPv6 then you're SOOL.

**Comment by medexer** [comment]
Please how can i fix this error log: FATAL:  remaining connection slots are reserved for non-replication superuser connections

**Comment by Afuhnwi-Afriitech** [comment]
each time i try connecting to postgresql via supabase edge, i get the error ConnectionParamsError: Missing connection parameters: database, user. here is my database url postgresql://postgres:xxxxxxxxxxxx@db.teuvqgwnaocxxxxxxyq.supabase.co:5432/postgres

**Reply by joobisb** [reply]
check if your password has special characters, if so encode them
